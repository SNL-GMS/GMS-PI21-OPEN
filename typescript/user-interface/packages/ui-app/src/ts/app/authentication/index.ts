import { ianAuthenticator } from './ian-authentication';
import type { Authenticator } from './types';

export const authenticator: Authenticator = ianAuthenticator;
