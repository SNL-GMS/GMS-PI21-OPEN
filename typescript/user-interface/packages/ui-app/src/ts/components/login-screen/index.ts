export { ReduxLoginScreenContainer as LoginScreen } from './login-screen-container';
