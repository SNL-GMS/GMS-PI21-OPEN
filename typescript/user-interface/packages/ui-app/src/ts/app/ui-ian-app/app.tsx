import { Intent, NonIdealState, Spinner } from '@blueprintjs/core';
import { IconNames } from '@blueprintjs/icons';
import { Displays } from '@gms/common-model';
import { IanDisplays } from '@gms/common-model/lib/displays/types';
import { IS_MODE_IAN } from '@gms/common-util';
import { getStore, useUiTheme, withReduxProvider } from '@gms/ui-state';
import * as React from 'react';
import { Provider } from 'react-redux';
import { HashRouter, Route, Routes } from 'react-router-dom';

import { AnalystUiComponents } from '~analyst-ui/';
import {
  IANMap,
  SignalDetectionsComponent,
  StationPropertiesComponent
} from '~analyst-ui/components';
import { FilterManager } from '~analyst-ui/components/filters/filter-manager';
import { InteractionConsumer } from '~analyst-ui/interactions/interaction-consumer';
import { InteractionProvider } from '~analyst-ui/interactions/interaction-provider';
import { authenticator } from '~app/authentication';
import { EffectiveNowTimeInitializer } from '~app/initializers/effective-time-now-initializer';
import { UIThemeWrapper } from '~app/initializers/ui-theme-wrapper';
import { ServiceWorkerController } from '~app/service-worker-controller';
import { ThemedToastContainer } from '~app/themed-toast-container';
import { CommonCommandRegistrar } from '~components/common-ui/commands';
import { CommandPalette } from '~components/common-ui/components/command-palette';
import { LoadingScreen } from '~components/loading-screen';
import { LoginScreen } from '~components/login-screen';
import { ProtectedRoute } from '~components/protected-route';
import { GoldenLayoutContext, Workspace } from '~components/workspace';
import { GMS_DISABLE_KEYCLOAK_AUTH } from '~env';

import { createPopoutComponent } from '../create-popout-component';
import { glContextData } from './golden-layout-config';

const store = getStore();

/**
 * Wraps the component route (not for SOH).
 * Provides the required context providers to the component.
 *
 * @param Component the component route
 * @param props the props passed down from the route to the component
 * @param suppressPopinIcon true to force suppress the golden-layout popin icon
 */
function wrap(Component: any, props: any, suppressPopinIcon = false) {
  function WrappedComponent(p) {
    const [uiTheme] = useUiTheme();
    return (
      <ServiceWorkerController>
        <>
          <UIThemeWrapper key={uiTheme.name}>
            <CommandPalette>
              {/* eslint-disable-next-line react/jsx-props-no-spreading */}
              <Component {...p} />
              <InteractionProvider>
                <InteractionConsumer />
              </InteractionProvider>
              <CommonCommandRegistrar />
            </CommandPalette>
            <ThemedToastContainer />
          </UIThemeWrapper>
          <EffectiveNowTimeInitializer />
          <FilterManager />
        </>
      </ServiceWorkerController>
    );
  }

  return createPopoutComponent(withReduxProvider(WrappedComponent), props, suppressPopinIcon);
}

const WrappedWorkflow: React.FC<unknown> = props => wrap(AnalystUiComponents.Workflow, props);
const WrappedEvents: React.FC<unknown> = props => wrap(AnalystUiComponents.Events, props);
const WrappedFilters: React.FC<unknown> = props => wrap(AnalystUiComponents.Filters, props);
const WrappedWaveformDisplay: React.FC<unknown> = props =>
  wrap(AnalystUiComponents.WaveformDisplay, props);
const WrappedMap: React.FC<unknown> = props => wrap(IANMap, props);
const WrappedSignalDetections: React.FC<unknown> = props => wrap(SignalDetectionsComponent, props);
const WrappedStationProperties: React.FC<unknown> = props =>
  wrap(StationPropertiesComponent, props);
const WrappedWorkspace: React.FC<unknown> = props => wrap(Workspace, props, true);

export function App(): React.ReactElement {
  // TODO: either re-route or make this a <404 component>
  const redirectRoute = GMS_DISABLE_KEYCLOAK_AUTH ? (
    <>
      <Route path="/login/:redirectUrl" element={<LoginScreen authenticator={authenticator} />} />
      <Route path="/login" element={<LoginScreen authenticator={authenticator} />} />
    </>
  ) : (
    <Route path="/:redirectUrl" />
  );

  return !IS_MODE_IAN ? (
    <NonIdealState
      icon={IconNames.ERROR}
      action={<Spinner intent={Intent.DANGER} />}
      title="Invalid settings"
      description="Not configured for IAN mode - Please check settings"
    />
  ) : (
    <Provider store={store}>
      <HashRouter>
        {
          // ! CAUTION: when changing the route paths
          // The route paths must match the `golden-layout` component name for popout windows
          // For example, the component name `my-route` must have the route path of `my-route`
        }
        {
          // For performance use `render` which accepts a functional component
          // that won't get unnecessarily remounted like with component.
        }
        <Routes>
          {
            // Authentication
            // React router dropped support for optional params
            // so we need separate routes for login and login with a redirect
          }
          {redirectRoute}
          <Route
            path="/loading"
            element={
              <ProtectedRoute redirectPath="loading" title="GMS: Loading...">
                <LoadingScreen />
              </ProtectedRoute>
            }
          />
          {
            // Individual ProtectedRoutes
          }
          <Route
            path={`/${IanDisplays.WORKFLOW}`}
            element={
              <ProtectedRoute
                redirectPath={IanDisplays.WORKFLOW}
                title={Displays.toDisplayTitle(Displays.IanDisplays.WORKFLOW, 'GMS: ')}
              >
                <WrappedWorkflow />
              </ProtectedRoute>
            }
          />
          <Route
            path={`/${IanDisplays.EVENTS}`}
            element={
              <ProtectedRoute
                redirectPath={IanDisplays.EVENTS}
                title={Displays.toDisplayTitle(Displays.IanDisplays.EVENTS, 'GMS: ')}
              >
                <WrappedEvents />
              </ProtectedRoute>
            }
          />
          <Route
            path={`/${IanDisplays.FILTERS}`}
            element={
              <ProtectedRoute
                redirectPath={IanDisplays.FILTERS}
                title={Displays.toDisplayTitle(Displays.IanDisplays.FILTERS, 'GMS: ')}
              >
                <WrappedFilters />
              </ProtectedRoute>
            }
          />
          <Route
            path={`/${IanDisplays.WAVEFORM}`}
            element={
              <ProtectedRoute
                redirectPath={IanDisplays.WAVEFORM}
                title={Displays.toDisplayTitle(Displays.IanDisplays.WAVEFORM, 'GMS: ')}
              >
                <WrappedWaveformDisplay />
              </ProtectedRoute>
            }
          />
          <Route
            path={`/${IanDisplays.MAP}`}
            element={
              <ProtectedRoute
                redirectPath={IanDisplays.MAP}
                title={Displays.toDisplayTitle(Displays.IanDisplays.MAP, 'GMS: ')}
              >
                <WrappedMap />
              </ProtectedRoute>
            }
          />
          <Route
            path={`/${IanDisplays.SIGNAL_DETECTIONS}`}
            element={
              <ProtectedRoute
                redirectPath={IanDisplays.SIGNAL_DETECTIONS}
                title={Displays.toDisplayTitle(Displays.IanDisplays.SIGNAL_DETECTIONS, 'GMS: ')}
              >
                <WrappedSignalDetections />
              </ProtectedRoute>
            }
          />
          <Route
            path={`/${IanDisplays.STATION_PROPERTIES}`}
            element={
              <ProtectedRoute
                redirectPath={IanDisplays.STATION_PROPERTIES}
                title={Displays.toDisplayTitle(Displays.IanDisplays.STATION_PROPERTIES, 'GMS: ')}
              >
                <WrappedStationProperties />
              </ProtectedRoute>
            }
          />
          {
            // Workspace
          }
          <Route
            path="*"
            element={
              <ProtectedRoute title="GMS: App Workspace">
                <GoldenLayoutContext.Provider value={glContextData()}>
                  <WrappedWorkspace />
                </GoldenLayoutContext.Provider>
              </ProtectedRoute>
            }
          />
        </Routes>
      </HashRouter>
    </Provider>
  );
}
