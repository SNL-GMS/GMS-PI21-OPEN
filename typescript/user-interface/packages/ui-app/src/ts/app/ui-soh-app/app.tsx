import { Intent, NonIdealState, Spinner } from '@blueprintjs/core';
import { IconNames } from '@blueprintjs/icons';
import { IS_MODE_SOH } from '@gms/common-util';
import {
  getStore,
  SohStatusSubscription,
  SystemMessageSubscription,
  withReduxProvider
} from '@gms/ui-state';
import * as React from 'react';
import { Provider } from 'react-redux';
import { HashRouter, Route, Routes } from 'react-router-dom';

import { authenticator } from '~app/authentication';
import { UIThemeWrapper } from '~app/initializers/ui-theme-wrapper';
import { ThemedToastContainer } from '~app/themed-toast-container';
import { SystemMessageAlertPlayer } from '~common-ui/components/system-message/audio';
import { CommonUIComponents } from '~components/common-ui';
import { CommonCommandRegistrar } from '~components/common-ui/commands';
import { CommandPalette } from '~components/common-ui/components/command-palette';
import { SohCommandRegistrar } from '~components/data-acquisition-ui/commands';
import { SohEnvironmentHistory } from '~components/data-acquisition-ui/components/environment-history';
import { SohLagHistory } from '~components/data-acquisition-ui/components/soh-lag-history';
import { SohMissingHistory } from '~components/data-acquisition-ui/components/soh-missing-history';
import { SohTimelinessHistory } from '~components/data-acquisition-ui/components/soh-timeliness-history';
import { StationStatistics } from '~components/data-acquisition-ui/components/station-statistics';
import { InteractionWrapper } from '~components/data-acquisition-ui/interactions/interaction-wrapper';
import { LoadingScreen } from '~components/loading-screen';
import { LoginScreen } from '~components/login-screen';
import { ProtectedRoute } from '~components/protected-route';
import { GoldenLayoutContext, Workspace } from '~components/workspace';
import { SohLag, SohMissing, SohTimeliness } from '~data-acquisition-ui/components/soh-bar-chart';
import { SohEnvironment } from '~data-acquisition-ui/components/soh-environment';
import { SohMap } from '~data-acquisition-ui/components/soh-map';
import { SohOverview } from '~data-acquisition-ui/components/soh-overview';

import { createPopoutComponent } from '../create-popout-component';
import { glContextData } from './golden-layout-config';

/**
 * Wraps the component route (not for SOH).
 * Provides the required context providers to the component.
 *
 * @param Component the component route
 * @param props the props passed down from the route to the component
 * @param suppressPopinIcon true to force suppress the golden-layout popin icon
 */
function wrap(Component: any, props: any, suppressPopinIcon = false) {
  const InteractionComp = InteractionWrapper(Component);

  function WrappedComponent(p) {
    return (
      <CommandPalette>
        {/* eslint-disable-next-line react/jsx-props-no-spreading */}
        <InteractionComp {...p} />
        <CommonCommandRegistrar />
        <ThemedToastContainer />
        <SystemMessageAlertPlayer />
      </CommandPalette>
    );
  }

  return createPopoutComponent(
    withReduxProvider(
      SystemMessageSubscription.wrapSystemMessageSubscription(WrappedComponent, props)
    ),
    props,
    suppressPopinIcon
  );
}

/**
 * Wraps the component route for SOH.
 * Provides the required context providers to the component.
 *
 * @param Component the component route
 * @param props the props passed down from the route to the component
 * @param suppressPopinIcon true to force suppress the golden-layout popin icon
 */
function wrapSoh(Component: any, props: any, suppressPopinIcon = false) {
  const InteractionComp = InteractionWrapper(Component);

  function WrappedComponent(p) {
    return (
      <UIThemeWrapper>
        <CommandPalette>
          {/* eslint-disable-next-line react/jsx-props-no-spreading */}
          <InteractionComp {...p} />
          <SohCommandRegistrar />
          <CommonCommandRegistrar />
        </CommandPalette>
        <ThemedToastContainer />
        <SystemMessageAlertPlayer />
      </UIThemeWrapper>
    );
  }
  return createPopoutComponent(
    withReduxProvider(
      SystemMessageSubscription.wrapSystemMessageSubscription(
        SohStatusSubscription.wrapSohStatusSubscription(WrappedComponent, props),
        props
      )
    ),
    props,
    suppressPopinIcon
  );
}

const WrappedSystemMessage: React.FC = props => wrap(CommonUIComponents.SystemMessage, props);
const WrappedOverview: React.FC = props => wrapSoh(SohOverview, props);
const WrappedStatistics: React.FC = props => wrapSoh(StationStatistics, props);
const WrappedEnvironment: React.FC = props => wrapSoh(SohEnvironment, props);
const WrappedMissing: React.FC = props => wrapSoh(SohMissing, props);
const WrappedLag: React.FC = props => wrapSoh(SohLag, props);
const WrappedEnvironmentTrends: React.FC = props => wrapSoh(SohEnvironmentHistory, props);
const WrappedLagTrends: React.FC = props => wrapSoh(SohLagHistory, props);
const WrappedMissingTrends: React.FC = props => wrapSoh(SohMissingHistory, props);
const WrappedTimelinessTrends: React.FC = props => wrapSoh(SohTimelinessHistory, props);
const WrappedTimeliness: React.FC = props => wrapSoh(SohTimeliness, props);
const WrappedMap: React.FC = props => wrapSoh(SohMap, props);
const WrappedWorkspace: React.FC = props => wrapSoh(Workspace, props, true);

export function App(): React.ReactElement {
  return !IS_MODE_SOH ? (
    <NonIdealState
      icon={IconNames.ERROR}
      action={<Spinner intent={Intent.DANGER} />}
      title="Invalid settings"
      description="Not configured for SOH mode - Please check settings"
    />
  ) : (
    <Provider store={getStore()}>
      <HashRouter>
        {
          // ! CAUTION: when changing the route paths
          // The route paths must match the `golden-layout` component name for popout windows
          // For example, the component name `signal-detections` must have the route path of `signal-detections`
        }
        {
          // For performance use `render` which accepts a functional component
          // that won't get unnecessarily remounted like with component.
        }
        <Routes>
          {
            // Authentication
            // React router dropped support for optional params
            // so we need separate routes for login and login with a redirect
          }
          <Route
            path="/login/:redirectUrl"
            element={<LoginScreen authenticator={authenticator} />}
          />
          <Route path="/login" element={<LoginScreen authenticator={authenticator} />} />
          <Route
            path="/loading"
            element={
              <ProtectedRoute>
                <LoadingScreen />
              </ProtectedRoute>
            }
          />
          {
            // Common UI
          }
          <Route
            path="/system-messages"
            element={
              <ProtectedRoute>
                <WrappedSystemMessage />
              </ProtectedRoute>
            }
          />
          {
            // Data Acquisition
          }
          <Route
            path="/soh-overview"
            element={
              <ProtectedRoute>
                <WrappedOverview />
              </ProtectedRoute>
            }
          />
          <Route
            path="/station-statistics"
            element={
              <ProtectedRoute>
                <WrappedStatistics />
              </ProtectedRoute>
            }
          />
          <Route
            path="/soh-environment"
            element={
              <ProtectedRoute>
                <WrappedEnvironment />
              </ProtectedRoute>
            }
          />
          <Route
            path="/soh-missing"
            element={
              <ProtectedRoute>
                <WrappedMissing />
              </ProtectedRoute>
            }
          />
          <Route
            path="/soh-lag"
            element={
              <ProtectedRoute>
                <WrappedLag />
              </ProtectedRoute>
            }
          />
          <Route
            path="/soh-environment-trends"
            element={
              <ProtectedRoute>
                <WrappedEnvironmentTrends />
              </ProtectedRoute>
            }
          />
          <Route
            path="/soh-lag-trends"
            element={
              <ProtectedRoute>
                <WrappedLagTrends />
              </ProtectedRoute>
            }
          />
          <Route
            path="/soh-missing-trends"
            element={
              <ProtectedRoute>
                <WrappedMissingTrends />
              </ProtectedRoute>
            }
          />
          <Route
            path="/soh-timeliness-trends"
            element={
              <ProtectedRoute>
                <WrappedTimelinessTrends />
              </ProtectedRoute>
            }
          />
          <Route
            path="/soh-timeliness"
            element={
              <ProtectedRoute>
                <WrappedTimeliness />
              </ProtectedRoute>
            }
          />
          <Route
            path="/soh-map"
            element={
              <ProtectedRoute>
                <WrappedMap />
              </ProtectedRoute>
            }
          />
          {
            // Workspace
          }
          <Route
            path="*"
            element={
              <ProtectedRoute>
                <GoldenLayoutContext.Provider value={glContextData()}>
                  <WrappedWorkspace />
                </GoldenLayoutContext.Provider>
              </ProtectedRoute>
            }
          />
        </Routes>
      </HashRouter>
    </Provider>
  );
}
