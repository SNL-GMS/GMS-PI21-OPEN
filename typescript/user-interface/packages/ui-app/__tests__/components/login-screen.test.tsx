/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';

import { authenticator } from '../../src/ts/app/authentication';
import { LoginScreenComponent } from '../../src/ts/components/login-screen/login-screen-component';
import type { LoginScreenReduxProps } from '../../src/ts/components/login-screen/types';

// eslint-disable-next-line @typescript-eslint/no-var-requires, @typescript-eslint/no-require-imports
const Enzyme = require('enzyme');

const reduxProps: LoginScreenReduxProps = {
  redirectPath: '',
  authenticated: false,
  authenticationCheckComplete: false,
  failedToConnect: true,
  setAppAuthenticationStatus: jest.fn()
};

// set up window alert and open so we don't see errors
(window as any).alert = jest.fn();
(window as any).open = jest.fn();
describe('Login screen', () => {
  it('should be defined', () => {
    expect(LoginScreenComponent).toBeDefined();
  });

  const loginScreenNonIdealState: any = Enzyme.mount(
    <LoginScreenComponent authenticator={authenticator} {...reduxProps} />
  );

  it('failed to connect should return non ideal state', () => {
    expect(loginScreenNonIdealState).toMatchSnapshot();
  });

  reduxProps.failedToConnect = false;

  const loginScreenNonIdealState2: any = Enzyme.mount(
    <LoginScreenComponent authenticator={authenticator} {...reduxProps} />
  );

  it('failed authentication check complete should return non ideal state', () => {
    expect(loginScreenNonIdealState2).toMatchSnapshot();
  });

  reduxProps.authenticationCheckComplete = true;
  reduxProps.authenticated = false;

  const loginScreen: any = Enzyme.mount(
    <LoginScreenComponent authenticator={authenticator} {...reduxProps} />
  );

  it('Connected, authentication check complete, and not authenticated should return login page', () => {
    expect(loginScreen).toMatchSnapshot();
  });

  const loginScreen2: any = Enzyme.shallow(
    <LoginScreenComponent authenticator={authenticator} {...reduxProps} />
  );

  loginScreen2.instance().setAppAuthenticationStatus({
    userName: 'someUser',
    authenticated: false,
    authenticationCheckComplete: true,
    failedToConnect: jest.fn()
  });
  // eslint-disable-next-line @typescript-eslint/unbound-method, jest/no-standalone-expect
  expect(reduxProps.setAppAuthenticationStatus).toHaveBeenCalled();

  reduxProps.failedToConnect = true;
  const loginScreen3: any = Enzyme.shallow(
    <LoginScreenComponent authenticator={authenticator} {...reduxProps} />
  );

  reduxProps.authenticated = true;
  reduxProps.authenticationCheckComplete = true;
  reduxProps.failedToConnect = false;

  const loginRedirect: any = Enzyme.shallow(
    <LoginScreenComponent authenticator={authenticator} {...reduxProps} />
  );

  it('Authenticated and accessing login page should return redirect', () => {
    expect(loginRedirect).toMatchSnapshot();
  });

  it('Private methods work as expected', async () => {
    loginScreen3.instance().state.username = 'someUser';
    const input = { key: 'Enter', stopPropagation: jest.fn() };
    loginScreen3.instance().stopPropagationAndLogin(input);
    expect(input.stopPropagation).toHaveBeenCalled();

    loginScreen3.instance().reconnectTimerId = undefined;
    loginScreen3.instance().forceUpdate();
    loginScreen3.instance().reconnect();
    // eslint-disable-next-line @typescript-eslint/unbound-method
    expect(reduxProps.setAppAuthenticationStatus).toHaveBeenCalled();

    await loginScreen3.instance().reconnectIfNotAuthenticated();
    // eslint-disable-next-line @typescript-eslint/unbound-method
    expect(reduxProps.setAppAuthenticationStatus).toHaveBeenCalled();

    const authStatus: any = { failedToConnect: true, authenticated: true };
    loginScreen3.instance().ifFailedToConnectTryAgain(authStatus);
    // eslint-disable-next-line
    expect(loginScreen3.instance().reconnectTimerId).toBeUndefined;

    authStatus.failedToConnect = false;
    loginScreen3.instance().ifFailedToConnectTryAgain(authStatus);
    // eslint-disable-next-line @typescript-eslint/unbound-method
    expect(reduxProps.setAppAuthenticationStatus).toHaveBeenCalled();

    loginScreen3.instance().setAppAuthenticationStatus(authStatus);
    loginScreen3.instance().setState = jest.fn();
    loginScreen3.instance().updateState({ target: { value: 'someUsername' } });
    expect(loginScreen3.instance().setState).toHaveBeenCalled();
  });

  it('React methods work as expected', () => {
    loginScreen3.instance().reconnectTimerId = 'someId';
    loginScreen3.instance().componentWillUnmount();
    expect(loginScreen3.instance().reconnectTimerId).toBeUndefined();
  });
});
