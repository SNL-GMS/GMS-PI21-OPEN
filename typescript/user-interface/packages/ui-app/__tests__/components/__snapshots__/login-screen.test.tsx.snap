// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login screen Authenticated and accessing login page should return redirect 1`] = `
<Navigate
  to={
    Object {
      "pathname": "/",
    }
  }
/>
`;

exports[`Login screen Connected, authentication check complete, and not authenticated should return login page 1`] = `
<LoginScreenComponent
  authenticated={false}
  authenticationCheckComplete={true}
  authenticator={
    Object {
      "authenticateWith": [Function],
      "checkIsAuthenticated": [Function],
      "logout": [Function],
      "unAuthenticateWith": [Function],
    }
  }
  failedToConnect={false}
  redirectPath=""
  setAppAuthenticationStatus={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "authenticated": false,
            "authenticationCheckComplete": true,
            "failedToConnect": [MockFunction],
            "userName": "someUser",
          },
        ],
        Array [
          Object {
            "authenticated": false,
            "authenticationCheckComplete": true,
            "failedToConnect": false,
            "userName": "",
          },
        ],
        Array [
          Object {
            "authenticated": false,
            "authenticationCheckComplete": true,
            "failedToConnect": false,
            "userName": "",
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <div
    className="bp4-dark login-screen-body"
  >
    <div
      className="login-container"
    >
      <form>
        <img
          alt=""
          id="GMSLogo"
          src="test-file-stub"
          width={370}
        />
        <div
          className="login-legal"
        />
        <div
          className="login-row user"
        >
          <div
            className="login-label"
          >
            Username:
          </div>
          <div
            className="login-input"
          >
            <Blueprint4.InputGroup
              autoFocus={true}
              className="login-input"
              data-cy="username-input"
              onChange={[Function]}
              onKeyDown={[Function]}
              type="text"
              value=""
            >
              <div
                className="bp4-input-group login-input"
              >
                <input
                  autoFocus={true}
                  className="bp4-input"
                  data-cy="username-input"
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  style={
                    Object {
                      "paddingLeft": undefined,
                      "paddingRight": undefined,
                    }
                  }
                  type="text"
                  value=""
                />
              </div>
            </Blueprint4.InputGroup>
          </div>
        </div>
        <div
          className="login-row password"
        >
          <div
            className="login-label"
          >
            Password:
          </div>
          <div
            className="login-input"
          >
            <Blueprint4.InputGroup
              autoComplete="off"
              className="login-input"
              disabled={true}
              onKeyDown={[Function]}
              type="password"
            >
              <div
                className="bp4-input-group bp4-disabled login-input"
              >
                <input
                  autoComplete="off"
                  className="bp4-input"
                  disabled={true}
                  onKeyDown={[Function]}
                  style={
                    Object {
                      "paddingLeft": undefined,
                      "paddingRight": undefined,
                    }
                  }
                  type="password"
                />
              </div>
            </Blueprint4.InputGroup>
          </div>
        </div>
        <div
          className="login-row login-button"
        >
          <Blueprint4.Button
            data-cy="login-btn"
            disabled={false}
            onClick={[Function]}
            text="Login"
          >
            <button
              className="bp4-button"
              data-cy="login-btn"
              onBlur={[Function]}
              onClick={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              type="button"
            >
              <Blueprint4.Icon
                aria-hidden={true}
                key="leftIcon"
                tabIndex={-1}
              />
              <span
                className="bp4-button-text"
                key="text"
              >
                Login
              </span>
              <Blueprint4.Icon
                key="rightIcon"
              />
            </button>
          </Blueprint4.Button>
        </div>
      </form>
    </div>
  </div>
  <ToastContainer
    autoClose={4000}
    closeButton={[Function]}
    closeOnClick={true}
    draggable={true}
    draggableDirection="x"
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnFocusLoss={true}
    pauseOnHover={true}
    position="bottom-right"
    role="alert"
    rtl={false}
    theme="dark"
    transition={[Function]}
  >
    <div
      className="Toastify"
    />
  </ToastContainer>
</LoginScreenComponent>
`;

exports[`Login screen failed authentication check complete should return non ideal state 1`] = `
<LoginScreenComponent
  authenticated={false}
  authenticationCheckComplete={false}
  authenticator={
    Object {
      "authenticateWith": [Function],
      "checkIsAuthenticated": [Function],
      "logout": [Function],
      "unAuthenticateWith": [Function],
    }
  }
  failedToConnect={false}
  redirectPath=""
  setAppAuthenticationStatus={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "authenticated": false,
            "authenticationCheckComplete": true,
            "failedToConnect": [MockFunction],
            "userName": "someUser",
          },
        ],
        Array [
          Object {
            "authenticated": false,
            "authenticationCheckComplete": true,
            "failedToConnect": false,
            "userName": "",
          },
        ],
        Array [
          Object {
            "authenticated": false,
            "authenticationCheckComplete": true,
            "failedToConnect": false,
            "userName": "",
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <Blueprint4.NonIdealState
    action={
      <Blueprint4.Spinner
        intent="primary"
      />
    }
    description="Attempting to login..."
    title="Checking login"
  >
    <div
      className="bp4-non-ideal-state"
    >
      <Component>
        <h4
          className="bp4-heading"
        >
          Checking login
        </h4>
      </Component>
      <div>
        Attempting to login...
      </div>
      <Blueprint4.Spinner
        intent="primary"
      >
        <div
          className="bp4-spinner bp4-intent-primary"
          role="progressbar"
        >
          <div
            className="bp4-spinner-animation"
          >
            <svg
              height={50}
              strokeWidth="8.00"
              viewBox="1.00 1.00 98.00 98.00"
              width={50}
            >
              <path
                className="bp4-spinner-track"
                d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
              />
              <path
                className="bp4-spinner-head"
                d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
                pathLength={280}
                strokeDasharray="280 280"
                strokeDashoffset={210}
              />
            </svg>
          </div>
        </div>
      </Blueprint4.Spinner>
    </div>
  </Blueprint4.NonIdealState>
</LoginScreenComponent>
`;

exports[`Login screen failed to connect should return non ideal state 1`] = `
<LoginScreenComponent
  authenticated={false}
  authenticationCheckComplete={false}
  authenticator={
    Object {
      "authenticateWith": [Function],
      "checkIsAuthenticated": [Function],
      "logout": [Function],
      "unAuthenticateWith": [Function],
    }
  }
  failedToConnect={true}
  redirectPath=""
  setAppAuthenticationStatus={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "authenticated": false,
            "authenticationCheckComplete": true,
            "failedToConnect": [MockFunction],
            "userName": "someUser",
          },
        ],
        Array [
          Object {
            "authenticated": false,
            "authenticationCheckComplete": true,
            "failedToConnect": false,
            "userName": "",
          },
        ],
        Array [
          Object {
            "authenticated": false,
            "authenticationCheckComplete": true,
            "failedToConnect": false,
            "userName": "",
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <Blueprint4.NonIdealState
    action={
      <Blueprint4.Spinner
        intent="danger"
      />
    }
    description="Attempting to connect..."
    icon="error"
    title="No connection to server"
  >
    <div
      className="bp4-non-ideal-state"
    >
      <div
        className="bp4-non-ideal-state-visual"
      >
        <Blueprint4.Icon
          aria-hidden={true}
          icon="error"
          size={60}
          tabIndex={-1}
        >
          <span
            aria-hidden={true}
            className="bp4-icon bp4-icon-error"
            icon="error"
            tabIndex={-1}
          >
            <svg
              data-icon="error"
              height={60}
              viewBox="0 0 20 20"
              width={60}
            >
              <path
                d="M10 0C4.48 0 0 4.48 0 10s4.48 10 10 10 10-4.48 10-10S15.52 0 10 0zm1 16H9v-2h2v2zm0-3H9V4h2v9z"
                fillRule="evenodd"
                key="0"
              />
            </svg>
          </span>
        </Blueprint4.Icon>
      </div>
      <Component>
        <h4
          className="bp4-heading"
        >
          No connection to server
        </h4>
      </Component>
      <div>
        Attempting to connect...
      </div>
      <Blueprint4.Spinner
        intent="danger"
      >
        <div
          className="bp4-spinner bp4-intent-danger"
          role="progressbar"
        >
          <div
            className="bp4-spinner-animation"
          >
            <svg
              height={50}
              strokeWidth="8.00"
              viewBox="1.00 1.00 98.00 98.00"
              width={50}
            >
              <path
                className="bp4-spinner-track"
                d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
              />
              <path
                className="bp4-spinner-head"
                d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
                pathLength={280}
                strokeDasharray="280 280"
                strokeDashoffset={210}
              />
            </svg>
          </div>
        </div>
      </Blueprint4.Spinner>
    </div>
  </Blueprint4.NonIdealState>
</LoginScreenComponent>
`;
